AmazonChimeSDK/AmazonChimeSDKTests/audiovideo/video/backgroundfilter/BackgroundFilterTest.swift

```diff
@@ -61,23 +67,60 @@ class BackgroundFilterTests: XCTestCase {
         }
 
         let videoSinkMock = mock(VideoSink.self)
-        var hash = ""
+        var processedImage: UIImage?
+        var videoFrameReceivedExpectation: XCTestExpectation?
 
-        given(videoSinkMock.onVideoFrameReceived(frame: any())) ~> { videoFrame in
-            hash = self.generateHash(frame: videoFrame)
+        given(videoSinkMock.onVideoFrameReceived(frame: any())) ~> { [self] videoFrame in
+            let result = self.processImage(frame: videoFrame)
+            processedImage = result.image
+            videoFrameReceivedExpectation!.fulfill()
         }
 
-        let backgroundBlurConfigurations = BackgroundBlurConfiguration(logger: ConsoleLogger(name: "BackgroundBlurProcessor"))
+        let backgroundBlurConfigurations = BackgroundBlurConfiguration(
+            logger: ConsoleLogger(name: "testBackgroundBlurVideoFrameProcessor"))
         let processor = BackgroundBlurVideoFrameProcessor(backgroundBlurConfiguration: backgroundBlurConfigurations)
 
         processor.addVideoSink(sink: videoSinkMock)
 
-        // Verfiy the checksum for the three different blur strengths.
-        let blurList = [BackgroundBlurStrength.low, BackgroundBlurStrength.medium, BackgroundBlurStrength.high]
+        // Verify the checksum for the different blur strengths.
+        let blurList = [BackgroundBlurStrength.high]
         for index in 0...(blurList.count - 1) {
+            videoFrameReceivedExpectation = expectation(description: "Video frame is received for index \(index)")
+
             processor.setBlurStrength(newBlurStrength: blurList[index])
             processor.onVideoFrameReceived(frame: frame)
-            XCTAssert(expectedBlurHashes[index].contains(hash), "Failed with \(hash) received hash.")
+
+            // Wait for the image to generate before proceeding to avoid non-determinism.
+            waitForExpectations(timeout: 1) { error in
```

Comment:
why not `wait([videoFrameReceivedExpectation], timeout: )`?

Reply:
no particular reason except I was perhaps lazy :) Yours is better, thanks!